src/bootstrap.rs

```diff
@@ -0,0 +1,319 @@
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+// available at: http://www.maidsafe.net/licenses
+//
+// Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied.
+//
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe Software.
+
+use time;
+use cbor;
+use cbor::CborTagEncode;
+use rustc_serialize::{Decodable, Decoder, Encodable, Encoder};
+use sodiumoxide::crypto;
+use transport::Endpoint;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path;
+use std::env;
+use std::cmp;
+use std::fmt;
+
+type BootStrapContacts = Vec<Contact>;
+
+static MAX_LIST_SIZE: usize = 1500;
+
+macro_rules! convert_to_array {
+    ($container:ident, $size:expr) => {{
+        if $container.len() != $size {
+            None
+        } else {
+            use std::mem;
+            let mut arr : [_; $size] = unsafe { mem::uninitialized() };
+            for element in $container.into_iter().enumerate() {
+                let old_val = mem::replace(&mut arr[element.0], element.1);
+                unsafe { mem::forget(old_val) };
+            }
+            Some(arr)
+        }
+    }};
+}
+
+#[derive(Clone)]
+pub enum PublicKey {
+    Asym(crypto::asymmetricbox::PublicKey),
+    Sign(crypto::sign::PublicKey),
+}
+
+impl cmp::PartialEq for PublicKey {
+    fn eq(&self, other: &PublicKey) -> bool {
+        match *self {
+            PublicKey::Asym(key0) => {
+                match *other {
+                    PublicKey::Asym(key1) => key0.0.iter().zip(key1.0.iter()).all(|a| a.0 == a.1),
+                    _ => false,
+                }
+            },
+            PublicKey::Sign(key0) => {
+                match *other {
+                    PublicKey::Sign(key1) => key0.0.iter().zip(key1.0.iter()).all(|a| a.0 == a.1),
+                    _ => false
+                }
+            },
+        }
+    }
+}
+
+impl fmt::Debug for PublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            PublicKey::Asym(key) => {
+                write!(f, "crypto::asymmetricbox::PublicKey {:?}", key.0.to_vec())
+            },
+            PublicKey::Sign(key) => {
+                write!(f, "crypto::sign::PublicKey {:?}", key.0.to_vec())
+            },
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Contact {
+    endpoint: Endpoint,
+    public_key: PublicKey,
```

Comment:
We will be needing both Asym(key) and Sign(key) in contact. 
This can be addressed later as we will not be using these in current scope.